pkg/plugin/connector/connutils/schema.go

```diff
@@ -38,6 +40,7 @@ func NewSchemaService(logger log.CtxLogger, registry schemaregistry.Registry) *S
 	return &SchemaService{
 		registry: registry,
 		logger:   logger.WithComponent("connutils.SchemaService"),
+		token:    uuid.NewString(),
```

Comment:
Looks like we create a global token for all connectors here. However, we said the token would be used to identify the connector that called the connector utility. This will be useful for logging, but also for things like namespacing when creating schemas (to prevent naming conflicts) or in a storage service in the future to keep data between connectors separate.

In my mind I imagined something like this:
- There should be some service that is tracking running connector plugins.
  - Maybe we can add that directly in `pkg/plugin/connector.PluginService`, although not sure if that would create circular dependencies.
- When creating a new dispenser, the registry would create a token specific for that connector and store it in a map of running connectors. That map allows us to know which token is used by which connector.
  - Or the dispenser could get a reference to `connector.PluginService` and it can register and deregister itself when it starts and when it stops. Registering would create the token and add it to the map of running connectors, deregistering would remove it from that map. That might be easier, since the dispenser actually knows when the plugin is started/stopped.
- The connector utility services (like this `SchemaService`) would also have a reference to the `connector.PluginService` so it can validate the token and get the info about which connector the token belongs to.

That's the high-level flow, the devil is in the detail of course, how to structure the code without circular dependencies. The point is that we'll end up with a place where we know what connector plugins are currently running and which token belongs to which plugin. The connector utilities can use that mapping to figure out if a token is valid and which connector sent the request.

Another approach would be to create a JWT token for each connector plugin so that the connector ID is stored in the token itself. This way we wouldn't need that mapping in a central place, since we don't have a different use case for that mapping right now other than connector utilities.

Reply:
Yup, you're right about the static token, and for this PR it was actually intentional, because it's enough to limit the usage of the schema service to the connectors. I'm about to implement proper authentication with tokens when handling name conflicts.